include/z64.h

```diff
@@ -1488,93 +1525,105 @@ struct ActorContext {
+    /* 0x00 */ u8   seqIndex;
+    /* 0x01 */ u8   nightSeqIndex;
+    /* 0x02 */ u8   unk_02;
+} SoundContext; // size = 0x3
+
+typedef struct {
+    /* 0x00 */ s32 enabled;
+    /* 0x04 */ s32 timer;
+} FrameAdvanceContext; // size = 0x8
+
+typedef struct {
+    /* 0x00 */ u16 state;
+} GameOverContext; // size = 0x02
+
 struct GlobalContext {
     /* 0x00000 */ GameState state;
     /* 0x000A4 */ s16 sceneNum;
-    /* 0x000A6 */ u8 sceneConfig; // TODO: This at least controls the behavior of animated textures. Does it do more?
-    /* 0x000A7 */ UNK_TYPE1 padA7[0x9];
-    /* 0x000B0 */ SceneCmd* currentSceneVram;
-    /* 0x000B4 */ UNK_TYPE1 padB4[0x4];
+    /* 0x000A6 */ u8 sceneConfig;
+    /* 0x000A7 */ char unk_A7[0x9];
+    /* 0x000B0 */ void* sceneSegment;
+    /* 0x000B4 */ char unk_B4[0x4];
     /* 0x000B8 */ View view;
-    /* 0x00220 */ Camera activeCameras[4];
+    /* 0x00220 */ Camera mainCamera;
+    /* 0x00398 */ Camera subCameras[3];
     /* 0x00800 */ Camera* cameraPtrs[4];
     /* 0x00810 */ s16 activeCamera;
-    /* 0x00812 */ s16 unk812;
-    /* 0x00814 */ u8 unk814;
-    /* 0x00815 */ u8 unk815;
-    /* 0x00816 */ UNK_TYPE1 pad816[0x2];
+    /* 0x00812 */ s16 nextCamera;
+    /* 0x00814 */ SoundContext soundCtx;
     /* 0x00818 */ LightContext lightCtx;
-    /* 0x00828 */ u32 unk828;
-    /* 0x0082C */ UNK_TYPE1 pad82C[0x4];
+    /* 0x00828 */ FrameAdvanceContext frameAdvCtx;
     /* 0x00830 */ CollisionContext colCtx;
     /* 0x01CA0 */ ActorContext actorCtx;
     /* 0x01F24 */ CutsceneContext csCtx;
-    /* 0x01F74 */ CutsceneEntry* cutsceneList;
     /* 0x01F78 */ SoundSource soundSources[16];
-    /* 0x02138 */ EffFootmark footmarks[100];
-    /* 0x046B8 */ SramContext sram;
-    /* 0x046D8 */ UNK_TYPE1 pad46D8[0x8];
+    /* 0x02138 */ EffFootmark footprintInfo[100];
+    /* 0x046B8 */ SramContext sramCtx;
     /* 0x046E0 */ SkyboxContext skyboxCtx;
     /* 0x04908 */ MessageContext msgCtx;
-    /* 0x169E0 */ UNK_TYPE1 pad169E0[0x8];
     /* 0x169E8 */ InterfaceContext interfaceCtx;
     /* 0x16D30 */ PauseContext pauseCtx;
-    /* 0x17000 */ u16 unk17000;
-    /* 0x17002 */ UNK_TYPE1 pad17002[0x2];
-    /* 0x17004 */ KankyoContext kankyoContext;
+    /* 0x17000 */ GameOverContext gameOverCtx;
+    /* 0x17004 */ EnvironmentContext envCtx;
     /* 0x17104 */ AnimationContext animationCtx;
     /* 0x17D88 */ ObjectContext objectCtx;
-    /* 0x186E0 */ RoomContext roomContext;
-    /* 0x18760 */ TransitionContext transitionCtx;
+    /* 0x186E0 */ RoomContext roomCtx;
+    /* 0x18760 */ DoorContext doorCtx;
     /* 0x18768 */ void (*playerInit)(Player* player, struct GlobalContext* globalCtx, FlexSkeletonHeader* skelHeader);
     /* 0x1876C */ void (*playerUpdate)(Player* player, struct GlobalContext* globalCtx, Input* input);
-    /* 0x18770 */ UNK_TYPE1 pad18770[0x8];
+    /* 0x18770 */ void* unk_18770; //! @TODO: Determine function prototype
+    /* 0x18774 */ s32 (*startPlayerFishing)(struct GlobalContext* globalCtx);
```

Comment:
```suggestion
    /* 0x18774 */ s32 (*startPlayerFishing)(struct GlobalContext* globalCtx);
```
Only OoT has fishing, not MM. What does this name mean then?

Reply:
MM does have Fishing, it's just unused.

Reply:
correct, although I think less remnants of it exist in ntsc-u, but in the japanese version its almost fully functional still and can be restored, see: https://www.youtube.com/watch?v=80a8G7Rsgoc

Reply:
`ovl_En_Fishing,46,4445,16764,364.43`
Very much looks like the whole actor's there: it's not much smaller than OoT debug's.